gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/store/setWriterMultisig.ts

```diff
@@ -0,0 +1,129 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+import { SolanaCommand, TransactionResponse, RawTransaction } from '@chainlink/gauntlet-solana'
+import { AccountMeta, PublicKey, Transaction, TransactionInstruction } from '@solana/web3.js'
+import { utils } from '@project-serum/anchor'
+import { CONTRACT_LIST, getContract } from '../../../lib/contracts'
+import { getRDD } from '../../../lib/rdd'
+
+type Input = {
+  transmissions: string
+  store: string
+}
+
+export default class SetWriterMultisig extends SolanaCommand {
+  static id = 'multisig:store:set_writer'
+  static category = CONTRACT_LIST.STORE
+
+  static examples = [
+    'yarn gauntlet store:set_writer --network=devnet --state=EPRYwrb1Dwi8VT5SutS4vYNdF8HqvE7QwvqeCCwHdVLC --ocrState=EPRYwrb1Dwi8VT5SutS4vYNdF8HqvE7QwvqeCCwHdVLC',
+  ]
+
+  constructor(flags, args) {
+    super(flags, args)
+
+    this.require(!!this.flags.ocrState, 'Please provide flags with "ocrState"')
+  }
+
+  makeInput = (userInput): Input => {
+    if (userInput) return userInput as Input
+    const rdd = getRDD(this.flags.rdd)
+    const agg = rdd.contracts[this.flags.ocrState]
+    return {
+      transmissions: agg.transmissionsAccount,
+      store: agg.storeAccount,
+    }
+  }
+
+  makeRawTransaction = async (signer: PublicKey) => {
+    const store = getContract(CONTRACT_LIST.STORE, '')
+    const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
+    const storeAddress = store.programId.toString()
+    const ocr2Address = ocr2.programId.toString()
+    const storeProgram = this.loadProgram(store.idl, storeAddress)
+    const ocr2Program = this.loadProgram(ocr2.idl, ocr2Address)
+
+    const input = this.makeInput(this.flags.input)
+    // const owner = this.wallet.payer
+
+    const storeState = new PublicKey(input.store || this.flags.state)
+    const ocr2State = new PublicKey(this.flags.ocrState)
+    const feedState = new PublicKey(input.transmissions)
+
+    const [storeAuthority, _storeNonce] = await PublicKey.findProgramAddress(
+      [Buffer.from(utils.bytes.utf8.encode('store')), ocr2State.toBuffer()],
+      ocr2Program.programId,
+    )
+
+    console.log(`Setting store writer on Store (${storeState.toString()}) and Feed (${feedState.toString()})`)
+
+    // const tx = await storeProgram.rpc.setWriter(storeAuthority, {
+    //   accounts: {
+    //     store: storeState,
+    //     feed: feedState,
+    //     authority: owner.publicKey,
+    //   },
+    //   signers: [owner],
+    // })
+
+    // logger.success(`Set writer on tx ${tx}`)
+    const data = storeProgram.coder.instruction.encode('set_writer', {
+      writer: storeAuthority
+    })
+
+    const accounts: AccountMeta[] = [
```

Comment:
Getting error code 0xbc2 -> 3010 -> AccountNotSigner, "The given account did not sign"